lib/io/hardware_i2c.hpp

```diff
@@ -2,10 +2,7 @@
 
 #include "i2c.hpp"
 
-#include <optional>
-
 #include <core/logger.hpp>
-#include <core/types.hpp>
```

Comment:
this file uses `core::Result` which comes from the header; wouldn't say it's unused

why do we care? well, i assume `core::Result` comes from one of the other headers, say `i2c.hpp`. what happens if `i2c.hpp` doesn't need `core/types.hpp` anymore? removing the import will cause errors in lots of other files and lead to very weird diffs. so you should import the things that are used in this file (even if the imports don't do anything, technically).

Reply:
thats fair, will add it back